### PR TITLE
Rails-Blog:  Add Devise

### PR DESCRIPTION
**Pull request by ZikaZaki**
# Rails-Blog:  Add Devise
## In this pull-request we did the followings:
- [x] Removed `current_user` method in `ApplicationController`, devise will provide us with one.
- [x] [Installed and setup devise](https://github.com/heartcombo/devise#getting-started).
   - Can register a new user.
   - User logs in with a combination of email and password.
   - Hashed password should be stored in the database.
   - Ask for confirmation of email.
   - Can reset password.
- [x] [Modified the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.
login
signup
- [x] Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- [x] Made sure that I used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- [x] Made sure that I documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- [x] Made sure to follow the list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,10 +1,14 @@
 class User < ApplicationRecord
-  has_many :posts, foreign_key: 'user_id'
-  has_many :comments, foreign_key: 'user_id'
-  has_many :likes, foreign_key: 'user_id'
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (elisha2kyakpo1) [inline app/models/user.rb]: - [x] In order to have all the functionality work, some attributes need to be present on the devise list. ` :confirmable` is one of the attributes that will help with user email confirmation.

File: db/migrate/20230106200813_add_devise_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      # t.datetime :current_sign_in_at
+      # t.datetime :last_sign_in_at
+      # t.string   :current_sign_in_ip
+      # t.string   :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
```
Comment (elisha2kyakpo1) [inline db/migrate/20230106200813_add_devise_to_users.rb]: - [ ] The confirmation of email works when these lines are uncommented, users should be asked to confirm their emails before they are saved.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description look good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20230106200813_add_devise_to_users.rb (hunk 2 above)
Comment (ZikaZaki) [reply db/migrate/20230106200813_add_devise_to_users.rb]: Thanks @elisha2kyakpo1  for the constructive feedback. I have already implemented all the requested changes. Thanks again for your time.

**Review by elmar8287 — APPROVED** [review]
Hi, @ZikaZaki.

# APPROVED  👍 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Congratulations! 🎉 

### Highlights

- All suggestions are implemented 👍 

### Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
### Please, remember to tag me (@elmar8287) in your question so I can receive the notification 👍 
giphy

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**